### PR TITLE
KF-1005 fix relation error

### PR DESCRIPTION
**Pull request by misohu**
- Check for minio connection exception (add waiting status when occured)
- Add update_status event handler to recover while waiting for minio

### CODE REVIEW

File: .github/workflows/integrate.yaml [hunk 1]
```diff
@@ -63,7 +63,7 @@ jobs:
         provider: microk8s
         channel: 1.24/stable
         # Pinned until this bug is resolved: https://bugs.launchpad.net/juju/+bug/1992833
-        bootstrap-options: "--agent-version=2.9.34"
+        bootstrap-options: "--agent-version=2.9.34 --model-default update-status-hook-interval='30s'"
```
Comment (i-chvets) [inline .github/workflows/integrate.yaml:66]: I understand we decrease update interval to speed up test. My concern about not using standard, real-life timing of 5 minutes comes from MySQL integration work. In that case, workload that talks to the DB was started without DB config or with incorrect DB config and timed out at 2-3 minutes and update status at 5 mins interval properly caught this situation. In this case, can we confirm workload behaviour: timeout is not happening/crash is not happening beyond 30 seconds interval.

File: src/services/s3.py [hunk 2]
```diff
@@ -41,6 +41,8 @@ def check_if_bucket_accessible(self, bucket_name):
             return True
         except botocore.exceptions.ClientError:
             return False
+        except botocore.exceptions.EndpointConnectionError as err:
+            raise err
```
Comment (ca-scribner) [inline src/services/s3.py:45]: what does this addition do? iiuc, this catches and immediately re-raises the error, so why include it?

File: src/charm.py [hunk 3]
```diff
@@ -56,6 +57,7 @@ def __init__(self, *args):
             self.framework.observe(self.on[rel].relation_changed, self._on_event)
         self._create_service()
 
+        self.framework.observe(self.on.update_status, self._on_event)
```
[the remark this answers is not in the document]
Comment (ca-scribner) [reply src/charm.py:60]: We decided to do this for now so this is not buggy, but to talk about better ways in future.

**Review by ca-scribner — APPROVED** [review]

**Comment by misohu** [comment]
Merging issue with update_status tracted in https://github.com/canonical/mlflow-operator/issues/133

**Pull request merged**